### PR TITLE
Remove data frame aggregation and make Stream.Writer asynchronous

### PR DESCRIPTION
**Pull request by olix0r**
Frame "accumulation" (i.e. data aggregation) was a nice idea that helped an
early test, but its complexity (currently) outweights its utility. Furthermore,
writing _to_ a stream (i.e. from an application) leaves the user to instrument
their own flow-control/backpressure, since Stream.Writer is synchronous and
doesn't account for backpressure.

This change does the following:

- Make Stream.Writer asynchronous and specific to Frame types.
- Completely remove AsyncQueueStreamer, which exposed frame-aggregation as a core feature.
- Provide a default Stream.Reader implementation that is backed by a AsyncQueue[Frame].
- Provide a default Stream.Writer implementation that uses an AsyncQueue[Frame] internally.
- Remove parm.MinAccumFrame.
- Remove Netty4Stream, which was an implementation of AsyncQueueStreamer.
- Netty4StreamTransport now simply uses the default Stream() implementation, passing it an AsyncQueue[Frame].
- Lots of test changes to accomodate this.

### CODE REVIEW

File: router/h2/src/e2e/scala/io/buoyant/router/h2/ConcurrentStreamsEndToEndTest.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ class ConcurrentStreamsEndToEndTest
   with ClientServerHelpers {
 
   // Tunable parameters:
-  val concurrencies = Seq(2, 4, 8, 16, 32, 64, 256)
+  val concurrencies = Seq(8)
```
Comment (adleong) [inline router/h2/src/e2e/scala/io/buoyant/router/h2/ConcurrentStreamsEndToEndTest.scala]: just curious why?

File: router/h2/src/e2e/scala/io/buoyant/router/h2/LargeStreamEndToEndTest.scala [hunk 2]
```diff
@@ -3,58 +3,54 @@ package h2
...
-    val client = upstream(server.server)
-    try {
+    withClient(_ => Response(Status.Ok, writer)) { client =>
+      val elapsed = Stopwatch.start()
       val req = Request("http", Method.Get, "host", "/path", Stream.Nil)
       val rsp = await(client(req))
       assert(rsp.status == Status.Ok)
-      await(defaultWait * 100) {
-        testStream(reader(rsp.data), writer, LargeStreamLen, 16 * 1024)
+      await(defaultWait * 2) {
```
Comment (adleong) [inline router/h2/src/e2e/scala/io/buoyant/router/h2/LargeStreamEndToEndTest.scala]: ugh

File: router/h2/src/e2e/scala/io/buoyant/router/h2/ConcurrentStreamsEndToEndTest.scala (hunk 1 above)
Comment (olix0r) [inline router/h2/src/e2e/scala/io/buoyant/router/h2/ConcurrentStreamsEndToEndTest.scala]: to speed up e2e tests (i.e. in ci).  we can always add more, but it felt like overkill for CI.

File: router/h2/src/e2e/scala/io/buoyant/router/h2/LargeStreamEndToEndTest.scala (hunk 2 above)
Comment (olix0r) [inline router/h2/src/e2e/scala/io/buoyant/router/h2/LargeStreamEndToEndTest.scala]: yeah, I'm coming to accept that stress testing doesn't really belong in the CI flow (i.e. shouldn't block commits)..

**Comment by olix0r** [comment]
merged https://github.com/BuoyantIO/linkerd/commit/cc2290055cfd838eecb661a4654502b3bebc6c85

**Pull request closed without merging**